### DISCUSSION TITLE
Way to enable/allow white space & capitalization in table columns

### DISCUSSION THREAD

**Discussion by ghost** [littleK0i/SnowDDL#117] [General]
We are loading data into Snowflake using Azure Data Factory (ADF).
ADF does an automatic column mapping from source dataset to the Snowflake table using and **exact** column name matching.
We have no control over the column names and their casing in the source dataset.
Since SnowDDL always uppercases the identifiers and does not allow white space in them, we are unable to load the data into the tables created by SnowDDL.
Is there an option to enable or allow white space and different capitalizations in colum names?

**Comment by littleK0i** [comment]
@MaRo99382 , identifier naming restrictions are intentional. Without these restrictions you may fix a small problem in a moment, but account becomes very difficult to use and navigate over time. Imagine having hundreds of schemas, each with different and obscure naming convention.

But there are a few things we can do to work around it.

1. For case normalization, try setting parameter `QUOTED_IDENTIFIERS_IGNORE_CASE ` to `TRUE` on User level or Session level. All lowercase identifiers will be automatically translated into uppercase.
https://docs.snowflake.com/en/sql-reference/parameters#quoted-identifiers-ignore-case

2. Create a "sandbox" schema, grant ownership privileges to external tool and let it create its own tables. After that you may copy data into permanent tables with proper naming and transformations. Most likely you'll need to implement at least some transformations anyway, so renaming can be a part of it.

3. You may sink data to Blob storage instead of Snowflake and run `COPY INTO` command yourself targeting Blob storage. ADF loader most likely does the same thing, but you have no control over `COPY INTO` command. And it is a big problem, because eventually you may want to use some other options, and you won't be able to if you do not have full control.

**Comment by ghost** [comment]
Hi @littleK0i,
thanks for your reply! I was gone for a couple of days.
3. is also my preferred way. Unfortunately, I have no power over the extraction tool we have in use...
I have to test 1. Since ADF is doing the column comparisons internally tansparently to determine the mapping, I don't know if this parameter helps.
If it does not work I have also found a way to parameterize the column mapping in ADF, so I don't have to rely on the automatic column mapping of ADF which is case-sensitive.
